test/integration/integration_test.cc

```diff
@@ -384,14 +384,13 @@ TEST_P(IntegrationTest, TestSmuggling) {
     EXPECT_EQ("HTTP/1.1 400 Bad Request\r\ncontent-length: 0\r\nconnection: close\r\n\r\n",
               response);
   }
-  // Make sure unsupported transfer encodings are rejected, lest they be abused.
   {
     std::string response;
     const std::string request = "GET / HTTP/1.1\r\nHost: host\r\ntransfer-encoding: "
                                 "identity,chunked \r\ncontent-length: 36\r\n\r\n" +
                                 smuggled_request;
     sendRawHttpAndWaitForResponse(lookupPort("http"), request.c_str(), &response, false);
-    EXPECT_EQ("HTTP/1.1 501 Not Implemented\r\ncontent-length: 0\r\nconnection: close\r\n\r\n",
+    EXPECT_EQ("HTTP/1.1 400 Bad Request\r\ncontent-length: 0\r\nconnection: close\r\n\r\n",
```

Comment:
Do you know why this behavioral change? It's probably something to flag to users.

Reply:
This changed because it is now caught at the parser level (throwing an exception there causes an HTTP 400) rather than in the codec's code due to the presence of `transfer-encoding` and `content-length` in an HTTP request. From the http-parser commit log
```
Furthermore, 3.3.3 of RFC 7230 explicitly states that presence of both
`Transfer-Encoding` and `Content-Length` indicates the smuggling attack
and "ought to be handled as an error".
```
This brings the behavior to be more uniform with the two examples above, but can understand the behavioral change being a concern. Do you want me to see if there's something I can do in the codec to keep this as a 501? Likely involves a bit of special-case handling in the exception catch.

Reply:
Probably not a big deal if this only occurs when smuggling, thanks for the explanation.